go/private/actions/binary.bzl

```diff
@@ -51,29 +52,34 @@ def emit_binary(ctx, go_toolchain,
   executables = {}
   extension = "" # TODO: .exe on windows
 
+  for mode in common_modes:
+    executable = ctx.new_file(name + "." + mode_string(mode) + extension)
+    executables[mode_string(mode)] = executable
+    go_toolchain.actions.link(
+        ctx,
+        go_toolchain = go_toolchain,
+        library=golib,
+        mode=mode,
+        executable=executable,
+        gc_linkopts=gc_linkopts,
+        x_defs=x_defs,
+    )
+
   if default:
     executables["default"] = default
-
-  for mode in link_modes:
-    executable = ctx.new_file(name + "." + mode + extension)
-    executables[mode] = executable
-
-  for mode, executable in executables.items():
-    if mode == "default":
-        # work out what the default mode should be
-        if "race" in ctx.features:
-            mode = RACE_MODE
-        elif "static" in ctx.features:
-            mode = STATIC_MODE
-        else:
-            mode = NORMAL_MODE
-
+    # work out what the default mode should be
```

Comment:
Not sure I understand this part. This executable will be redundant with one of the files linked above, right? Would both be linked in the same build?

Also, can `get_mode` be used below? If not, nit: 2-space indent.

Reply:
There is no change in here. This is a consequence of not being able to specify executable outputs, which means you need to distinct files one that is always the "race" build, and one that is the executable output that changes based on the feature flags.